include/openenclave/edger8r/common.h

```diff
@@ -80,10 +80,10 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
  * Compute and set the pointer value for the given parameter within the input
  * buffer. Make sure that the buffer has enough space.
  */
-#define OE_SET_IN_POINTER(argname, argsize)                                  \
+#define OE_SET_IN_POINTER(argname, argsize, type)                            \
     if (pargs_in->argname)                                                   \
     {                                                                        \
-        *(uint8_t**)&pargs_in->argname = input_buffer + input_buffer_offset; \
+        *(type*)&(pargs_in->argname) = (type)(input_buffer + input_buffer_offset); \
```

Comment:
One of our goals has been that the generated code is C++ compatible since one day we would like to generate C++ files.
Can you make sure that this code is C++ compatible (I don't see any issues with it).
You can test C++ compatibility directly #including the generates C files in a C++ file

Reply:
```
#include "aliasing_t.c"

add_enclave(TARGET edl_aliasing_enc CXX SOURCES aliasing.cpp ${aliasing_t}pp)

[303/754] : && /usr/bin/c++  -g   tests/oeedger8r/enc/CMakeFiles/edl_aliasing_enc.dir/aliasing.cpp.o tests/oeedger8r/enc/CMakeFiles/edl_aliasing_enc.dir/aliasing_t.cpp.o  -o tests/oeedger8r/enc/edl_aliasing_enc ... suceeded
```

Looks like it worked fine.